### PR TITLE
Add tests for Treelite binary checkpoint files

### LINKED ISSUE #54: Add E2E test for Treelite checkpoint model
Depends on #49

### PR DESCRIPTION
**Pull request by wphicks**
Add tests to confirm that Treelite binary checkpoint files can be loaded and used in Triton
Provide scripts for converting pickled sklearn or cuML files
Update README with instructions for using sklearn or cuML RF models

Close #54

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,9 +87,76 @@ model_repository/
...
+directly supported by the Triton FIL backend, random forest models trained with
+[Scikit-Learn](https://scikit-learn.org/stable/modules/model_persistence.html)
+or [cuML](https://docs.rapids.ai/api/cuml/stable/pickling_cuml_models.html) are
+generally serialized using Python's
+[pickle](https://docs.python.org/3/library/pickle.html) module. In order to
+avoid a round-trip through Python in Triton, we instead provide scripts to
+convert pickled RF models to Treelite's binary checkpoint format. You can
+download the relevant script for Scikit-Learn [here](https://raw.githubusercontent.com/triton-inference-server/fil_backend/main/scripts/convert_sklearn)
+and for cuML
+[here](https://raw.githubusercontent.com/triton-inference-server/fil_backend/main/scripts/convert_cuml.py).
```
Comment (dantegd) [inline README.md:107]: Might be worth to immediately check links after merge to be sure they work well

File: qa/L0_e2e/generate_example_model.py [hunk 2]
```diff
@@ -133,6 +183,26 @@ def train_lightgbm_regressor(data, targets, depth=25, trees=100):
     return model
 
 
+def train_sklearn_regressor(data, targets, depth=25, trees=100):
```
Comment (dantegd) [inline qa/L0_e2e/generate_example_model.py:186]: Not a big deal and probably worth of a cleanup follow up PR, but these functions could be condensed to a single ones with the classifier ones if the class is passed as parameter

File: qa/L0_e2e/test_model.py [hunk 3]
```diff
@@ -280,22 +295,32 @@ def run_test(
...
+        model_path = os.path.join(model_dir, 'checkpoint.tl')
+        model_type = 'treelite_checkpoint'
     else:
         raise RuntimeError('Model format not recognized')
 
-    fil_model = cuml.ForestInference.load(
-        model_path, output_class=output_class, model_type=model_type
-    )
+    # NOTE: Once FIL has been updated to directly load treelite_checkpoint
+    # models, this workaround should be removed
```
Comment (dantegd) [inline qa/L0_e2e/test_model.py:315]: You’re probably tracking this in an issue already I think, but it’d be worth adding a link here

File: scripts/environment.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+channels:
+  - conda-forge
+  - nvidia
+  - rapidsai-nightly
+dependencies:
+  - cudatoolkit=11.0
+  - cuml=21.06
+  - python=3.8
+  - scikit-learn
+  - treelite
```
Comment (dantegd) [inline scripts/environment.yml:11]: Might be worth pinning the minimum version of treelite

**Review by dantegd — APPROVED** [review]
Had a few suggestions but the code looks good

File: README.md (hunk 1 above)
Comment (wphicks) [reply README.md:107]: Will do! Thanks for the reminder

File: qa/L0_e2e/generate_example_model.py (hunk 2 above)
Comment (wphicks) [reply qa/L0_e2e/generate_example_model.py:186]: Yeah, I think that makes good sense! @hcho3 expressed some interest in more general refactoring of these test scripts post-21.06, so if it's all right by you, I think I'll leave this for that effort.

Comment (wphicks) [reply qa/L0_e2e/generate_example_model.py:186]: Also added it as an issue so we don't forget: https://github.com/triton-inference-server/fil_backend/issues/72

**Pull request merged**